### PR TITLE
add support for fetch timeout control for node_helpers, fix timeouts on armv6l

### PR DESCRIPTION
**Pull request by sdetweil**
user reporting slow/no connection/timeout errors on armv6l for calendar, and newsfeed

we can increase the timeout by adding calls to the undici lib, but it requires node 20.18.1 or above.

this adds the support for timeout
(also environment variable to override if needed,, mmFetchTimeout (default 30 seconds)

and updates the base node version

### CODE REVIEW

**Comment by bugsounet** [comment]
Warn: package-lock

integrity/resolve missing

**Comment by khassel** [comment]
for changing the built-in node fetch timeout we have to add the whole `undici` http client as dependency?

I would prefer a simpler solutions e.g. adding env var `--network-family-autoselection-attempt-timeout` (untestet)

**Comment by sdetweil** [comment]
no idea what this drags in 

const { setGlobalDispatcher, Agent } = require("undici");

**Comment by bugsounet** [comment]
/!\ resolve and integrity are removed from package-lock.json

For information it's allows to download each libraries from an compressed file
Without it, MagicMiror deps will take many many Time for installing

**Comment by sdetweil** [comment]
@bugsounet  I don't know what you are telling me..  git says there are two lines changed in package.json

systeminfo(changed version)  and undici (added)

after I changed, I erased package-lock.json , and did a new npm install  to get the latest

**Comment by khassel** [comment]
@sdetweil how did you test this? With an arm v6 machine or can I simulate this some other way?

**Comment by sdetweil** [comment]
i tested on the armv6l that had the timeout error on both news and calendar and also on my amd64 machine.

**Comment by khassel** [comment]
what url did you use for testing/simulating?

**Comment by sdetweil** [comment]
the default config urls. on armv6l
and my google cal and default on my amd

i had hard coded it in calendar to
prove, news failed

thats why the question about fixing for any/all via app/whatever
then put this on the armv6, and both worked
and of course all the testcases e2e and electron

**Comment by sdetweil** [comment]
only way to simulate would be to create sone cpu sucker threads to cause delays . armv6 problem is small memory and slow processor handling swapping on sd card

**Comment by sdetweil** [comment]
i wonder if the guys 11 calendars failure would be fixed here

**Comment by khassel** [comment]
@sdetweil my question was only how I can test it, used [this not working url now](http://test-ipv6.coloradointerlink.net/).

Your solution works, my tests with `--network-family-autoselection-attempt-timeout` does not work. The `undici` package is small, so I'm fine with merging this beside the `package-lock.json` problem.

If you look at the diffs of `package-lock.json` you removed many many lines like

```json
			"resolved": "https://registry.npmjs.org/@altano/repository-tools/-/repository-tools-0.1.1.tgz",
			"integrity": "sha512-5vbUs2A98CC3g1AlOBdkBE0BMukkLjLIsMHAtuxg6Pt9dQXxYWdLKOf66v6c/vIqtNcgTMv0oGtddLdMuH9X6w==",
```

Don't know how you did it but this must be reverted before merge ...

**Comment by sdetweil** [comment]
@khassel  oh, how to test timeout 
use this

https://httpbin.org/delay/10

the number is seconds of delay

**Comment by sdetweil** [comment]
```
If you look at the diffs of package-lock.json you removed many many lines like
```
I did  NOT "remove" anything

I see the resolved is missing on some, and the integrity value is different.. I just ran npm install with no package-lock, and no node_modules folder to generate a new one , this IS a different node version.. 

npm install (not install-mm) on my amd64 is blazing fast,  <30 seconds total

**Comment by khassel** [comment]
> I did NOT "remove" anything

look at the diff

**Comment by sdetweil** [comment]
@khassel I saw the diff.. but I didn't DO it on purpose.. just npm install

**Comment by khassel** [comment]
seems to be a npm issue, maybe this helps: https://github.com/npm/cli/issues/4263#issuecomment-1173093952

**Comment by sdetweil** [comment]
I can npm install again without removing the files and it looks better..
pushed

**Comment by bugsounet** [comment]
Remove node_helper and do npm install again
Diff package-lock.json and see result.
Must be fine

**Comment by sdetweil** [comment]
package-lock.json did not change

**Comment by bugsounet** [comment]
I can't verify from my computer, sorry

**Comment by sdetweil** [comment]
>I can't verify from my computer, sorry
I don't know what you are telling me

**Comment by bugsounet** [comment]
@sdetweil : verified it's ok for me :)

File: package.json [hunk 1]
```diff
@@ -105,7 +106,7 @@
 		"electron": "^32.2.7"
 	},
 	"engines": {
-		"node": ">=20.9.0 <21 || >=22"
+		"node": ">=20.18.1 <21 || 22 || >=23"
```
Comment (khassel) [inline package.json:109]: can you change this back to >=22 at the end? Thx.

Comment (bugsounet) [reply package.json]: But... engines.node is not fixed on package-lock

Comment (sdetweil) [reply package.json]: how I am I supposed to know all the steps?  I don't know what is is the lockfile except it always causes trouble..
posted an update but too late

Comment (khassel) [reply package.json]: will update dependencies before release and fix it ... I should not merge late at night ...

**Pull request merged**